### ISSUE TITLE
Question about deferred linking

### ISSUE THREAD

**Issue by DenizUgur** [gpac/gpac#2851]
I'm using the following flags to build a filter graph iteratively. I've observed that compared to normal operations fanout doesn't happen at the correct place. I feel like this is highly dependant on my implementation with gpac but if I could get an insight on it, that would be perfect.

|flag|
|---|
|GF_FS_FLAG_NON_BLOCKING|
|GF_FS_FLAG_NO_REASSIGN|
|GF_FS_FLAG_REQUIRE_SOURCE_ID|
|GF_FS_FLAG_FORCE_DEFER_LINK|
|GF_FS_FLAG_PREVENT_PLAY|

Basically I'm loading `fin` then two `reframer` consecutively. I then explicitly define the connections. The resulting graph has two `mp4dmx` going to reframers. I was expecting this fanout to happen after `mp4dmx`. I know I haven't given any implementation details but I just want to know if this is expected given the flags list above.

> After every API call to libgpac I'm running the session

**Comment by jeanlf** [comment]
Something like:

```
gpac -no-reassign -step -sid -np -dl -i av2.mp4 @ reframer @1 reframer -rl=2 -graph
```

This seems to produce the expected output. Coud you try giving more implementation details ?

**Comment by DenizUgur** [comment]
Take the following examples and their outputs. Second example is what I'm expecting but without explicitly connecting to the demuxer.

## Example 1

```ts
const fin = sfn.loadFilter(s, "fin:src=video.mp4", error)
await sfn.run(s)

const ref1 = sfn.loadFilter(s, "reframer", error)
await sfn.run(s)

const ref2 = sfn.loadFilter(s, "reframer", error)
await sfn.run(s)

ffn.setSource(ref1, fin, "V1")
ffn.reconnectOutput(fin, null)
await sfn.run(s)

ffn.setSource(ref2, fin, "V1")
ffn.reconnectOutput(fin, null)
await sfn.run(s)

session.printGraph()
```

```shell
Filters connected:
fin (src=/opfs/imported/3/video.mp4) (idx=1)
-(PID video.mp4) mp4dmx (dyn_idx=4)
--(PID V1) reframer (idx=2)
-(PID video.mp4) mp4dmx (dyn_idx=5)
--(PID V1) reframer (idx=2)
--(PID V1) reframer (idx=3)
```

## Example 2

```ts
const fin = sfn.loadFilter(s, "fin:src=video.mp4", error)
await sfn.run(s)

const ref1 = sfn.loadFilter(s, "reframer", error)
await sfn.run(s)

const ref2 = sfn.loadFilter(s, "reframer", error)
await sfn.run(s)

ffn.setSource(ref1, fin, "V1")
ffn.reconnectOutput(fin, null)
await sfn.run(s)

const ref1ipid = ffn.getInputPID(ref1, 0)
const mp4dmx = ffn.getSource(ref1ipid)

ffn.setSource(ref2, mp4dmx, "V1")
ffn.reconnectOutput(mp4dmx, null)
await sfn.run(s)

session.printGraph()
```

```shell
Filters connected:
fin (src=/opfs/imported/3/video.mp4) (idx=1)
-(PID video.mp4) mp4dmx (dyn_idx=4)
--(PID V1) reframer (idx=2)
--(PID V1) reframer (idx=3)
```

**Comment by jeanlf** [comment]
OK I could reproduce: the second relink was indeed loading a new mp4dmx to solve  fin->ref2 chain, now fixed on master.

I added a new syntax for links to be used in defer mode, where the link can be specified independently of the target filter (see `gpac -h defer`). So your first example can be tested using:

```
gpac -no-reassign -step -sid -np -dl -i video.mp4 -f reframer -f reframer @1@2 -rl=2 @0@2 -rl=2 -graph
```

**Comment by DenizUgur** [comment]
Confirmed the fix. Thank you @jeanlf
